src/Markdig/Extensions/Alerts/AlertInlineParser.cs

```diff
@@ -0,0 +1,126 @@
+using Markdig.Parsers;
+using Markdig.Renderers.Html;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace Markdig.Extensions.Alerts;
+
+/// <summary>
+/// An inline parser for an alert inline (e.g. `[!NOTE]`).
+/// </summary>
+/// <seealso cref="InlineParser" />
+public class AlertInlineParser : InlineParser
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AlertInlineParser"/> class.
+    /// </summary>
+    public AlertInlineParser()
+    {
+        OpeningCharacters = ['['];
+    }
+
+    public override bool Match(InlineProcessor processor, ref StringSlice slice)
+    {
+        // We expect the alert to be the first child of a quote block. Example:
+        // > [!NOTE]
+        // > This is a note
+        if (!(processor.Block is ParagraphBlock paragraphBlock) || !(paragraphBlock.Parent is QuoteBlock quoteBlock) || quoteBlock.Count != 1 || paragraphBlock.Inline?.FirstChild != null)
+        {
+            return false;
+        }
+
+        var startPosition = processor.GetSourcePosition(slice.Start, out int line, out int column);
+
+        var saved = slice;
+        var c = slice.NextChar();
+        if (c != '!')
+        {
+            slice = saved;
+            return false;
+        }
+
+        c = slice.NextChar(); // Skip !
+
+        var start = slice.Start;
+        var end = start;
+        while (c.IsAlphaUpper())
+        {
+            end = slice.Start;
+            c = slice.NextChar();
+        }
+
+        // We need at least one character
+        if (c != ']' || start == end)
+        {
+            slice = saved;
+            return false;
+        }
+
+        var alertType = new StringSlice(slice.Text, start, end);
+        c = slice.NextChar(); // Skip ]
+
+        start = slice.Start;
+        while (true)
+        {
+            if (c == '\0' || c == '\n' || c == '\r')
+            {
+                end = slice.Start;
+                if (c == '\r')
+                {
+                    c = slice.NextChar(); // Skip \r
+                    if (c == '\0' || c == '\n')
+                    {
+                        end = slice.Start;
+                        if (c == '\n')
+                        {
+                            slice.NextChar(); // Skip \n
+                        }
+                    }
+                }
+                else if (c == '\n')
+                {
+                    slice.NextChar(); // Skip \n
+                }
+                break;
+            }
+            else if (!c.IsSpaceOrTab()) 
+            {
+                slice = saved;
+                return false;
+            }
+
+            c = slice.NextChar();
+        }
+
+        var alertBlock = new AlertBlock(alertType)
+        {
+            Span = new SourceSpan(startPosition, processor.GetSourcePosition(slice.Start - 1)),
+            TriviaSpaceAfterKind = new StringSlice(slice.Text, start, end),
+            Line = line,
+            Column = column
```

Comment:
```suggestion
            Span = quoteBlock.Span,
            TriviaSpaceAfterKind = new StringSlice(slice.Text, start, end),
            Line = quoteBlock.Line,
            Column = quoteBlock.Column
```

I think we might want to inherit the positions of the quote block here, otherwise the first `> ` is skipped.

Reply:
Fixed via commit af38833